proxy/proxy/broker/rhnBroker.py

```diff
@@ -661,9 +661,12 @@ def __checkAuthSessionTokenCache(self, token, channel):
         self.cachedClientInfo['X-RHN-Server-ID'] = self.clientServerId
         log_debug(4, 'Retrieved token from cache: %s' % self.cachedClientInfo)
 
+        authChannels = [x[0] for x in self.authChannels]
+        log_debug(4, "Auth channels: '%s'" % authChannels)
+
         # Compare the two things
-        if not _dictEquals(token, self.cachedClientInfo,
-                           ['X-RHN-Auth-Channels']):
+        if not _dictEquals(token, self.cachedClientInfo, ['X-RHN-Auth-Channels']) or \
+                channel not in authChannels:
```

Comment:
I don't like the original syntax, sorry. I would suggest to change it this way:
```suggestion
        if (
            not _dictEquals(token, self.cachedClientInfo, ['X-RHN-Auth-Channels'])
            or channel not in authChannels
        ):
```